packages/discord.js/src/structures/MentionableSelectMenuInteraction.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const MessageComponentInteraction = require('./MessageComponentInteraction');
+
+/**
+ * Represents a {@link ComponentType.MentionableSelect} select menu interaction.
+ * @extends {MessageComponentInteraction}
+ */
+class MentionableSelectMenuInteraction extends MessageComponentInteraction {
+  constructor(client, data) {
+    super(client, data);
+
+    const { members, users, roles } = data.data.resolved || {};
+
+    /**
+     * Collection of the selected users
+     * @type {Collection<Snowflake, User>}
+     */
+    this.users = new Collection();
+
+    /**
+     * Collection of the selected users
+     * @type {Collection<Snowflake, GuildMember|APIGuildMember>}
+     */
+    this.members = new Collection();
```

Comment:
Shouldn't this be `null` if the interaction is from a dm based channel?

Reply:
Unsure, this was also a dispute in the old PR - I opted t make it never null. cc @vladfrangu

Reply:
If you want access the members and the interaction is from a dm based channel you get `GuildMember | APIGuildMember | undefined`. Wouldn't it be better to just make it `null` for dm based channels. You should use `.inGuild()` type guard anyway.

Reply:
Type-wise you'll always get `GuildMember | APIGuildMember | undefined` anyway, `Collection#get` returns `V | undefined`.

Reply:
Yes. I meant it looks like it is possible to get a `GuildMember` if the interaction is from a dm based channel, but that's not really the case.

Reply:
> Shouldn't this be `null` if the interaction is from a dm based channel?

No, making it `null` would barely serve a benefit here, and it would also drive down a rabbit hole where we would need *even more* type guards.

Besides, `Collection#get` already returns `undefined` on missing entries, so there isn't really much of a need to make the entire thing nullable.